internal/pkg/composable/providers/kubernetes/hints_test.go

```diff
@@ -366,3 +367,119 @@ func TestGenerateHintsMappingWithLogStream(t *testing.T) {
 
 	assert.Equal(t, expected, hintsMapping)
 }
+
+func TestGenerateHintsMappingWithProcessors(t *testing.T) {
+	logger := getLogger()
+	pod := &kubernetes.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "testpod",
+			UID:       types.UID(uid),
+			Namespace: "testns",
+			Labels: map[string]string{
+				"foo":        "bar",
+				"with-dash":  "dash-value",
+				"with/slash": "some/path",
+			},
+			Annotations: map[string]string{
+				"app": "production",
+			},
+		},
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		Spec: kubernetes.PodSpec{
+			NodeName: "testnode",
+		},
+		Status: kubernetes.PodStatus{PodIP: "127.0.0.5"},
+	}
+
+	mapping := map[string]interface{}{
+		"namespace": pod.GetNamespace(),
+		"pod": mapstr.M{
+			"uid":  string(pod.GetUID()),
+			"name": pod.GetName(),
+			"ip":   pod.Status.PodIP,
+		},
+		"namespace_annotations": mapstr.M{
+			"nsa": "nsb",
+		},
+		"labels": mapstr.M{
+			"foo":        "bar",
+			"with-dash":  "dash-value",
+			"with/slash": "some/path",
+		},
+		"annotations": mapstr.M{
```

Comment:
Can we have all the test inputs aligned for better clarity? For example the `annotations` field should include the package `apache` and annotations should also be present in the `Pod` sample above.

Reply:
Done see [89bc94d](https://github.com/elastic/elastic-agent/pull/3107/commits/89bc94da832e5f75279eb3133f629eade09a50c9)